### ISSUE TITLE
Setting an object that is not an association fails unless empty options are set

### ISSUE THREAD

**Issue by jrrbru** [rjz/backbone.associate#18]
If I try to set an attribute's value to a Backbone.Model object, and that attribute doesn't have an association, the value is never set. 

Lines 108-110 of `backbone.associate.js` appear to be causing a problem in some instances:
```javascript
if (_.isObject(val) && (typeof options === "undefined" || options === null)) {
  options = val;
}
```

This ends up setting my desired object value as the options, which is passed into Backbone's `set` method as the second parameter. Then, on line 464 of backbone.js, we get
```javascript
var unset      = options.unset;
```

Because options is a Backbone.Model, it has an `unset()` method and `unset` is truthy. Thus on line 491, my attribute value gets deleted from the hash of attributes to be set:
```javascript
unset ? delete current[attr] : current[attr] = val;
```

I'm not at all sure why lines 108-110 in `backbone.associate.js` even exist. Why would someone call `set` with no value, but rather with options in the value position? Can you explain the reason for this code so I can come up with a pull request to fix the issue?

**Comment by jrrbru** [comment]
BTW, here's a test to add to the spec file that fails to demonstrate the issue:
```javascript
      it('should accept an object as a value argument', function () {
        var modelB = new this.modelB();
        this.parent.set('b', modelB);
        expect(this.parent.attributes.b).toEqual(modelB);
      });
```

**Comment by jrrbru** [comment]
Thinking back over this, it occurs to me that perhaps the thought was that if an object is passed in the first parameter, then the second parameter should be options and not value. If that is the case, then a simple change to 
```javascript
 if (_.isObject(key) && (typeof options === "undefined" || options === null)) {
  options = val;
}
```
would fix the issue. This would check if the key is an object, rather than if the val is an object.

**Comment by rjz** [comment]
@jrrbru, thanks for the very clear report and thoughtful analysis!

I believe your suggested fix will address this issue without introducing any unexpected consequences—if you'd be willing to PR the regression test and bugfix, I'd be happy to merge them up.

**Comment by jrrbru** [comment]
Made pull request #19 to fix the bug.

**Comment by rjz** [comment]
Merged and pushed to npm as v0.0.11.
